src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.TranslationAction_Actions.cs

```diff
@@ -279,10 +323,14 @@ public override GeneratorDriver TransformGeneratorDriver(GeneratorDriver _)
                     return generatorDriver;
                 }
 
-                public override CompilationAndGeneratorDriverTranslationAction? TryMergeWithPrior(CompilationAndGeneratorDriverTranslationAction priorAction)
+                public override TranslationAction? TryMergeWithPrior(TranslationAction priorAction)
                 {
-                    // If the prior action is also a ReplaceGeneratorDriverAction, we'd entirely overwrite it's changes, so we can drop the prior one entirely.
-                    return priorAction is ReplaceGeneratorDriverAction ? this : null;
+                    // If the prior action is also a ReplaceGeneratorDriverAction, we'd entirely overwrite it's changes,
+                    // so we can drop the prior one's generator driver entirely.  Note: we still want to use it's
+                    // `OldProjectState` as that still represents the prior state we're translating from.
+                    return priorAction is ReplaceGeneratorDriverAction
+                        ? new ReplaceGeneratorDriverAction(oldGeneratorDriver, priorAction.OldProjectState, newProjectState)
+                        : null;
```

Comment:
this preserves the sematnics that after merging, our 'oldstate' was the original old state of the prior action.